test/mocks/token-util-mock.js

```diff
@@ -14,69 +14,69 @@
 const Q = require("q");
 
 function decode(tokenString) {
-	if (tokenString === "invalid_token") {
-		return undefined;
-	} else if (tokenString === "bad_scope") {
-		return {scope: "bad_scope"};
-	} else if (tokenString === "null_scope") {
-		return null;
-	} else if (tokenString === "access_token_mock_test_scope") {
-		return {scope: "test_scope"};
-	} else if (tokenString === "id_token_mock_test_scope") {
-		return {scope: "test_scope"};
-	} else {
-		return {scope: "appid_default"};
-	}
+  if (tokenString === "invalid_token") {
+	return undefined;
+  } else if (tokenString === "bad_scope") {
+	return {scope: "bad_scope"};
+  } else if (tokenString === "null_scope") {
+	return null;
+  } else if (tokenString === "access_token_mock_test_scope") {
+	return {scope: "test_scope"};
+  } else if (tokenString === "id_token_mock_test_scope") {
+	return {scope: "test_scope"};
+  } else {
+	return {scope: "appid_default"};
+  }
 }
 
 function decodeAndValidate(tokenString) {
-	let deferred = Q.defer();
-	if (tokenString === "invalid_token") {
-		deferred.resolve();
-	} else if (tokenString === "bad_scope") {
-		deferred.resolve({scope: "bad_scope"});
-	} else if (tokenString === "null_scope") {
-		deferred.resolve(null);
-	} else if (tokenString === "access_token_mock_test_scope") {
-		deferred.resolve({scope: "test_scope"});
-	} else if (tokenString === "id_token_mock_test_scope") {
-		deferred.resolve({scope: "test_scope"});
-	} else {
-		deferred.resolve({scope: "appid_default"});
-	}
-	return deferred.promise;
+  let deferred = Q.defer();
+  if (tokenString === "invalid_token") {
+	deferred.resolve();
+  } else if (tokenString === "bad_scope") {
+	deferred.resolve({scope: "bad_scope", audience: "myCliendId"});
```

Comment:
I think the audience field inside the token should be 'aud' instead of 'audience'